server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -80,12 +80,26 @@ private static boolean searchWithCollectorManager(
         try {
             final ReduceableSearchResult result = searcher.search(query, collectorManager);
             result.reduce(queryResult);
-        } catch (EarlyTerminatingCollector.EarlyTerminationException e) {
-            queryResult.terminatedEarly(true);
-        } catch (TimeExceededException e) {
+        } catch (RuntimeException re) {
+            // Rethrow exception if cause is null
+            if (re.getCause() == null) {
+                throw re;
+            }
+
+            // Unwrap the RuntimeException and ExecutionException from Lucene concurrent search method and rethrow
+            if (re.getCause() instanceof ExecutionException || re.getCause() instanceof InterruptedException) {
+                Throwable t = re.getCause();
+                if (t.getCause() != null) {
+                    rethrow(t.getCause());
+                }
+            }
+
+            // Rethrow any unexpected exception types
+            rethrow(re.getCause());
```

Comment:
```suggestion
rethrowCauseIfPossible(re);
```

Reply:
Can we instead wrap the cause in `QueryPhaseExecutionException` ? That way we don't have to deal with this casting here. Also there are multiple other places from which we throw this exception so I don't see any issue with double wrapping happening [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/query/QueryPhase.java#L305).

Reply:
I think that should work, I will double check that this doesn't change any of the exceptions we are expecting since it's technically adding another layer to the exceptions.